### PR TITLE
Add contrib to doc 

### LINKED ISSUE #346: DOC ENH add contribution instructions
We can reuse content from https://scikit-learn.org/stable/developers/contributing.html#how-to-contribute

### PR DESCRIPTION
**Pull request by bcharlier**
should fix #346

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -113,6 +114,7 @@
         ("CLI", "cli"),
         ("API", "api"),
         ("Write a benchmark", "how"),
+        # ("Contributing", "contrib"),
```
Comment (tomMoral) [inline doc/conf.py:117]: Leftover?

**Review by tomMoral — APPROVED** [review]
This is very neat, thx @bcharlier .
Just a leftover line I gues? then we can merge.

**Comment by agramfort** [comment]
Looks good https://output.circle-artifacts.com/output/job/adf822df-c0e2-4e3d-9929-8daad963c0ee/artifacts/0/dev/index.html#how-to-contribute

File: doc/contrib.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+   .. prompt:: bash $
+
+        git switch -c my_feature
+
+   and start making changes. Always use a feature branch. It's good
+   practice to never work on the ``main`` branch!
+
+
+8. Develop the feature on your feature branch on your computer, using Git to
```
Comment (mathurinm) [inline doc/contrib.rst:85]: This line appears in bold in the doc: 
image

File: doc/contrib.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    do the version control. When you're done editing, add changed files using
+    ``git add`` and then ``git commit``:
+
+    .. prompt:: bash $
+
+        git add modified_files
```
Comment (mathurinm) [inline doc/contrib.rst:91]:
```suggestion
        git add <your_modified_files>
```

File: doc/contrib.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+reStructuredText documents (like this one), tutorials, etc. reStructuredText
+documents live in the source code repository under the ``doc/`` directory.
+
+You can edit the documentation using any text editor, and then generate the
+HTML output by typing, in a shell:
+
+.. prompt:: bash $
+
+    cd doc/
+    make html-noplot                # make html
```
Comment (mathurinm) [inline doc/contrib.rst:127]:
```suggestion
    make html
```

File: doc/contrib.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+We are glad to accept any sort of documentation: function docstrings,
+reStructuredText documents (like this one), tutorials, etc. reStructuredText
+documents live in the source code repository under the ``doc/`` directory.
+
+You can edit the documentation using any text editor, and then generate the
+HTML output by typing, in a shell:
+
+.. prompt:: bash $
+
+    cd doc/
```
Comment (mathurinm) [inline doc/contrib.rst:126]:
```suggestion
    pip install benchopt[doc]
    cd doc/
```

**Comment by mathurinm** [comment]
Thank you @bcharlier! I did a few edit suggestions

File: doc/contrib.rst (hunk 2 above)
Comment (mathurinm) [reply doc/contrib.rst:84]: apparently it's because the line below was over indented. fixed

File: doc/conf.py (hunk 1 above)
Comment (bcharlier) [reply doc/conf.py:117]: yes, it can be safely removed...

**Comment by tomMoral** [comment]
Not sure why you closed the PR @bcharlier ?

**Review by mathurinm — APPROVED** [review]

**Pull request merged**